.github/run_erblint.sh

```diff
@@ -2,7 +2,15 @@
 
 shopt -s globstar
 
-bundle exec erblint decidim**/app/{cells,views}/**/*.erb
+CURRENT_BRANCH=$(git branch --show-current)
+DEFAULT_BRANCH=$(basename $(git symbolic-ref refs/remotes/origin/HEAD --short))
+
+ALL_VIEWS=decidim**/app/{cells,views}/**/*.erb
+MODIFIED_VIEWS=$(git diff --name-only "${DEFAULT_BRANCH}" | grep -E '(views|cells).erb$')
+
+# Scan only the modified ERB files, except for the default branch
+FILES=$([[ "${CURRENT_BRANCH}" == "${DEFAULT_BRANCH}" ]] && echo ${ALL_VIEWS} || echo ${MODIFIED_VIEWS})
+[[ -n "$FILES" ]] && bundle exec erblint ${FILES} || echo "No ERB files changed"
```

Comment:
When there are errors this doesn't work as expected, as it gives the errors and also it says that it doesn't changed. You can check it out by generating an error in a view and then running this: 

```bash
$ [[ true ]]  && bundle exec erblint decidim-core/app/views/decidim/shared/_share_modal.html.erb || echo "No ERB files changed"
 
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
warning: parser/current is loading parser/ruby31, which recognizes3.1.2-compliant syntax, but you are running 3.1.1.
Please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
Linting 1 files with 13 linters...

Extra whitespace detected at end of line.
In file: decidim-core/app/views/decidim/shared/_share_modal.html.erb:9

Extra whitespace detected at end of line.
In file: decidim-core/app/views/decidim/shared/_share_modal.html.erb:10

2 error(s) were found in ERB files
No ERB files changed

```

I think this needs to be changed to a good old conditional:

```suggestion

if [[ -n "$FILES" ]] ; then 
  bundle exec erblint ${FILES} 
else 
  echo "No ERB files changed"
fi
```

Reply:
Good catch. It turns out `bundle exec erblint` returns 1 as exit code when there are ERB errors. `bash` takes such exit as an error, therefore it runs the other part of the ternary operator.